pkg/apis/meta/v1alpha1/net.go

```diff
@@ -9,49 +10,95 @@ import (
 
 // PortMapping defines a port mapping between the VM and the host
 type PortMapping struct {
-	HostPort uint64 `json:"hostPort"`
-	VMPort   uint64 `json:"vmPort"`
+	BindAddress net.IP   `json:"bindAddress,omitempty"`
+	HostPort    uint64   `json:"hostPort"`
+	VMPort      uint64   `json:"vmPort"`
+	Protocol    Protocol `json:"protocol,omitempty"`
```

Comment:
I'm think that omitempty doesn't work with custom marshallers, but idk. We can test. If so, then we need to use a pointer here.

Reply:
Now that we're using the standard mashaler and Docker's parser always adding the proto, this isn't an issue.